<launch>
    
  <node pkg="rviz" type="rviz" name="rviz" cwd="node" required="false" output="screen" />

  <node pkg="image_view" type="image_view" name="image_view" cwd="node" required="false">
    <remap from="image" to="/rgbd_odometry/depth_matches"/>
    <param name="autosize" value="true" />
  </node>

 <node pkg="dynamic_reconfigure" type="reconfigure_gui" name="dynamic_reconfigure" cwd="node" required="false" output="screen" />

  <node pkg="rgbd_vo_waitinglist" type="main" name="rgbd_odometry" cwd="node" required="true" output="screen" >
    <param name="/feature_matcher/feature_detector"                 value="4"/>
    <param name="/feature_matcher/feature_extractor"                value="4"/>
    <param name="/feature_matcher/descriptor_matcher"                value="1"/>
    <param name="/feature_matcher/horizontal_matches"                 value="false"/>
    <param name="/feature_matcher/horizontal_threshold"               value="1"/>
    <param name="/feature_matcher/matching_distance_ratio_threshold"  value="0.8"/>
    <param name="/feature_matcher/max_radius_search_dist"             value="100"/>
    <param name="/feature_matcher/tracking_matches"                   value="false"/>
    <param name="/feature_matcher/tracking_threshold"                 value="120"/>
    <param name="/feature_matcher/distance_factor"                    value="1.0"/>
    <param name="/feature_matcher/feature_detector_parameter_name"    value="maxTotalKeypoints"/>
    <param name="/feature_matcher/feature_detector_parameter_value"  value="1000"/>
    <param name="/feature_matcher/feature_extractor_parameter_name"    value="orientationNormalized"/>
    <param name="/feature_matcher/feature_extractor_parameter_value"  value="1"/>
    <param name="/feature_matcher/timing_debug"  value="false"/>
    <param name="/feature_matcher/double_check_tracking_matches"  value="1"/>
    <param name="/feature_matcher/distinct_matches"  value="true"/>
    <param name="/feature_matcher/reduce_search_area"  value="true"/>
    <param name="/feature_matcher/search_distance"  value="30"/>

    <param name="/ransac_tranformation/ransac_max_distance_inliers"  value="0.02"/>
    <param name="/ransac_tranformation/ransac_iterations"  value="250"/>
  </node>


</launch>
